dbt/adapters/athena/impl.py

```diff
@@ -48,25 +91,25 @@ def s3_uuid_table_location(self):
         return f"{client.s3_staging_dir}tables/{str(uuid4())}/"
 
     @available
-    def clean_up_partitions(
-        self, database_name: str, table_name: str, where_condition: str
-    ):
+    def clean_up_partitions(self, database_name: str, table_name: str, where_condition: str):
         # Look up Glue partitions & clean up
         conn = self.connections.get_thread_connection()
         client = conn.handle
```

Comment:
dbt already has thread local connection, so we only need to use boto3 session created by PyAthena connection.

Reply:
So don't have to use locks.